openmls/src/treesync/mod.rs

```diff
@@ -225,6 +226,42 @@ impl TreeSync {
         }
     }
 
+    /// Create a [`TreeSync`] instance from a vector of nodes without expecting
+    /// there to be a [`KeyPackage`] that belongs to this particular MLS client.
+    /// WARNING: Some of the [`TreeSync`] invariants will not hold for this
+    /// tree, as the `own_leaf_index` does not point to a leaf with private key
+    /// material in it.
+    pub(crate) fn from_nodes_without_leaf(
+        backend: &impl OpenMlsCryptoProvider,
+        ciphersuite: &Ciphersuite,
+        node_options: &[Option<Node>],
```

Comment:
As long as the caller consumes this do we really need to use a slice here?
This function looks pretty expensive because you iterate over the entire tree (cloning and converting). We could keep it as is and file a follow up to figure out if it really is a performance issue if you prefer.

Reply:
Yes, this is very annoying. There are multiple functions handling these `Option<Node>` vectors or slices and we should be consistent. In the end, we invariably need to clone them to instantiate a tree, so we might as well use full vectors. I assume that in almost all cases, the caller doesn't need the nodes afterwards anyway. I chose the slice here, because we are using slices in pretty much all other places that we handle these tree copies. I'll file a follow-up to optimize it and if nothing else, make it consistent.

Reply:
👍🏻

Reply:
Filed #652.